perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_facility_attribute_def_virt_UIDRanges.java

```diff
@@ -0,0 +1,108 @@
+
+	private static final String A_E_namespaceUIDRanges = AttributesManager.NS_ENTITYLESS_ATTR_DEF + ":namespace-UIDRanges";
+	private static final String A_FAC_uidNamespace = AttributesManager.NS_FACILITY_ATTR_DEF + ":uid-namespace";
+
+	@Override
+	public void checkAttributeSemantics(PerunSessionImpl sess, Facility facility, Attribute attribute) throws InternalErrorException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
+		try {
+			Attribute uidNamespaceAttribute = getUidNamespaceAttribute(sess, facility);
+			if(uidNamespaceAttribute.getValue() == null) throw new WrongReferenceAttributeValueException(attribute, uidNamespaceAttribute, "There is missing UID namespace on the facility.");
+			Attribute namespaceUIDRangesAttribute = getNamespaceUIDRangesAttribute(sess, uidNamespaceAttribute.valueAsString());
+			sess.getPerunBl().getAttributesManagerBl().checkAttributeSemantics(sess, uidNamespaceAttribute.valueAsString(), namespaceUIDRangesAttribute);
+		} catch(WrongReferenceAttributeValueException ex) {
+			throw new WrongReferenceAttributeValueException(attribute, ex.getReferenceAttribute());
+		}
+
+	}
+
+	@Override
+	public Attribute fillAttribute(PerunSessionImpl sess, Facility facility, AttributeDefinition attributeDefinition) {
+		return new Attribute(attributeDefinition);
+	}
+
+	@Override
+	public Attribute getAttributeValue(PerunSessionImpl sess, Facility facility, AttributeDefinition attributeDefinition) throws InternalErrorException {
+		Attribute attribute = new Attribute(attributeDefinition);
+		Attribute uidNamespaceAttribute = getUidNamespaceAttribute(sess, facility);
+		if(uidNamespaceAttribute.getValue() == null) return attribute;
+		Attribute namespaceUIDRangesAttribute = getNamespaceUIDRangesAttribute(sess, uidNamespaceAttribute.valueAsString());
+		return Utils.copyAttributeToVirtualAttributeWithValue(namespaceUIDRangesAttribute, attribute);
+	}
+
+	@Override
+	public boolean setAttributeValue(PerunSessionImpl sess, Facility facility, Attribute attribute) throws InternalErrorException, WrongReferenceAttributeValueException {
+		Attribute uidNamespaceAttribute = getUidNamespaceAttribute(sess, facility);
+		if(uidNamespaceAttribute.getValue() == null) throw new WrongReferenceAttributeValueException(attribute, uidNamespaceAttribute, "There is missing UID namespace on the facility.");
+		Attribute namespaceUIDRangesAttribute = getNamespaceUIDRangesAttribute(sess, uidNamespaceAttribute.valueAsString());
+		if(! (attribute.getValue() == null ? namespaceUIDRangesAttribute.getValue() == null : attribute.getValue().equals(namespaceUIDRangesAttribute.getValue()))) {
+			//attribute from param have other vale then physical attribute
+			throw new WrongReferenceAttributeValueException(attribute, namespaceUIDRangesAttribute, "You can't change attribute value of UID Ranges by changing value of virtual attribute.");
+		}
+		return false;
+	}
```

Comment:
Maybe it is problem with gid ranges too, but I am not sure why is this logic here at all. You definitely can't set anything this way (there is no logic for setting in it) so maybe we should skip it at all. Or did I miss something?